### DISCUSSION TITLE
Sonoff TRVZB Temperature setting

### DISCUSSION THREAD

**Discussion by jamesonuk** [Koenkk/zigbee2mqtt#21693] [Devices]
Just trying to separate discussion around setting up the devices and the and the changes / 1.1.4 firmware.

Taken from https://github.com/Koenkk/zigbee2mqtt/issues/19269

**Comment by arthurdent2095** [comment]
I am using better thermostat and target temperature mode for now and getting good results in Normal mode.

However I note that this is sending _hundreds_ of updates per day, which presumably - if this is flash storage on the device -  is going to significantly shorten its life - is it ? Would be good to know the truth on that.

I tried using offset mode and the number of updates is significantly less, an order of magnitude less in fact. However, there is the 1.1.1 bug which I have only read about. But moreover, the offset from better thermostat seems to cap out at -2.5 degrees where the valve supposedly can handle -7 degrees offset. This means I don't get good results because the temperature error when heating is anything up to 7 degrees. I have not had time yet to experiment with software overrides/hacks to see if I can remove the 2.5 degree limitation. 

I am using ZHA. The limitation may be coming from there. I have not researched the overhead of migration to Z2M. 

EDIT: It looks like a change was merged into HA Core a week ago and so this may be fixed in HA Core 2024.3.0, I need to test it. 

I am using the demand signals from the underlying valves (not from BT) to drive a MHCOZY dry contact relay which is in parallel to an existing netatmo controller in order to bring the boiler on. The netatmo setup is not great there is some optimisation where the netatmo valves don't always close (which I thought I had worked around, but saw some data today that says no), so soon I expect to rip that out and go sonoff all the way, but really I won't do that until these teething issues are understood. 


Finally, as a control I am running one room without using better thermostat all, but with a zigbee temperature sensor recording the room itself, and I have also observed that although the radiator valve measured temperature over shoots and shuts off too early, on the second heating cycle the room gets to temperature and it manages to hold that temperate pretty well through the day. Just an observation that maybe I don't need all the fuss around BT and should just let the valve do its thing. Ymmv but maybe I should go back to vanilla valve usage one room at a time and if the data is comparable just leave it and chill.

**Comment by arthurdent2095** [comment]
Just confirming that after the 2024.3 release of HA core, better thermostat using ZHA is happily editing the temperature offsets to up to 7 degrees which means this mode is now usable as opposed to target temperature mode. The frequency of updates to the valves is now down in the tens per day as opposed to hundreds. This should improve the life of the electronics if we assume its a flash update each time.  

Edit: I am suffering from these TRVZB valves "crashing" or locking up somehow periodically. But in an odd way, whilst I get a "failed to deliver" message in ZHA logs on the one hand, the valves are still checking in periodically on the other.  If I flip them into auto mode by pressing the button, HA knows about it immediately. If I ask HA to send them a command (like modify set point, or update local temperature offset) it just times out with an Ember error. So it's like they are sending commands (I can change the set point manually on the valve!) but refusing to accept them over zigbee.

The only way I have found to solve this is to reset them by taking the batteries out and putting them back in, wait for the recalibrate, HA then finds them and they can be sent commands again. 

I do not know if this is another firmware issue but it must be. Would nice to know if anyone else has seen this on ZHA or Z2M in case its a ZHA problem ? 

I now have a monitor script every 3H checking for stuck valves. It's pretty good at finding them but rebooting them is going to get very tedious over time and the WAF for this is .... very low.

**Comment by rouke-broersma** [comment]
After some hours all 7 of my trvzb start adjusting the temperature by themselves between the temperature set in HA and an old setpoint (not at the same time, but all have this issue). Sometimes it's fixed by restarting z2m, or the zigbee coordinator. Most of the time I need to remove them from the network for some hours. This also locks up z2m due to message spam. HA and the trv seem to be fighting each other over the setpoint. The trvs also constantly open and close the valve due to this. Does anyone else experience this?

**Reply by photomoose** [reply]
Are your TRVs in the "Auto" system mode? If so, they will follow the schedule that has been programmed on the device (visible in the Exposes page in Z2M) and will conflict/fight with any temperature changes made in HA. If you set the system to mode to "Heat", the TRV will follow the requested temperature set point as requested by HA.

**Reply by rouke-broersma** [reply]
They are in Heat mode :/ I have also enabled child lock, but that does not seem to help either. I have never enabled Auto mode as far as I know. I have tried clearing the schedule as well just to be sure but the schedule does not seem to be allowed to be empty, and if I set a simple schedule after a while (or after a restart, not sure) the default schedule comes back (still not turning on Auto, just something I observed).
